compiler/src/main/java/com/xfinity/resourceprovider/RpCodeGenerator.java

```diff
@@ -50,38 +51,68 @@ TypeSpec generateResourceProviderClass() {
+                                           .addStatement("this.drawableProvider = new DrawableProvider(context)")
+                                           .addStatement("this.colorProvider = new ColorProvider(context)")
+                                           .addStatement("this.dimenProvider = new DimensionProvider(context)")
+                                           .addStatement("this.integerProvider = new IntegerProvider(context)")
                                            .build();
 
         ClassName stringProviderClassName = get(packageName, "StringProvider");
-        FieldSpec stringProvider = FieldSpec.builder(stringProviderClassName, "string")
-                                            .addModifiers(Modifier.PUBLIC, Modifier.FINAL).build();
+        FieldSpec stringProvider = FieldSpec.builder(stringProviderClassName, "stringProvider")
+                                            .addModifiers(Modifier.PRIVATE, Modifier.FINAL).build();
 
         ClassName drawableProviderClassName = get(packageName, "DrawableProvider");
-        FieldSpec drawableProvider = FieldSpec.builder(drawableProviderClassName, "drawable")
-                                              .addModifiers(Modifier.PUBLIC, Modifier.FINAL).build();
+        FieldSpec drawableProvider = FieldSpec.builder(drawableProviderClassName, "drawableProvider")
+                                              .addModifiers(Modifier.PRIVATE, Modifier.FINAL).build();
 
         ClassName colorProviderClassName = get(packageName, "ColorProvider");
-        FieldSpec colorProvider = FieldSpec.builder(colorProviderClassName, "color")
-                                           .addModifiers(Modifier.PUBLIC, Modifier.FINAL).build();
+        FieldSpec colorProvider = FieldSpec.builder(colorProviderClassName, "colorProvider")
+                                           .addModifiers(Modifier.PRIVATE, Modifier.FINAL).build();
 
         ClassName dimenProviderClassName = get(packageName, "DimensionProvider");
-        FieldSpec dimenProvider = FieldSpec.builder(dimenProviderClassName, "dimen")
-                                           .addModifiers(Modifier.PUBLIC, Modifier.FINAL).build();
+        FieldSpec dimenProvider = FieldSpec.builder(dimenProviderClassName, "dimenProvider")
+                                           .addModifiers(Modifier.PRIVATE, Modifier.FINAL).build();
 
         ClassName integerProviderClassName = get(packageName, "IntegerProvider");
-        FieldSpec integerProvider = FieldSpec.builder(integerProviderClassName, "integer")
-                                             .addModifiers(Modifier.PUBLIC, Modifier.FINAL).build();
-
+        FieldSpec integerProvider = FieldSpec.builder(integerProviderClassName, "integerProvider")
+                                             .addModifiers(Modifier.PRIVATE, Modifier.FINAL).build();
+
+        MethodSpec getStringMethodSpec = MethodSpec.methodBuilder("getString")
```

Comment:
Would consider naming all these getters with a plural, i.e. get Strings

Reply:
I named them this way so that the API, when used from kotlin, would mirror the R class.  

i.e.  rp.string.getSomeString()

thoughts?

Reply:
Either way is fine, really. One way is going to read weirdly in Java, the other I think is fine both ways. If you like it for the RP.string.whatever style, I'm alright with that.

Reply:
yeah, I can make them plural.  it's close enough

Reply:
fixed